tools/tests/package-tests.js

```diff
@@ -1952,3 +1958,637 @@ selftest.define("show package w/many versions",
     ]
   });
  });
+
+
+// This tests that we get the right excerpt out of the Readme.md in different
+// combinations. It doesn't test publication, because publishing is slow --
+// that's covered in a different test.
```

Comment:
I mean, there are definitely *unit* tests in self-test.  (Things that don't depend much on global initialization stuff.)  Like utils-tests.js and version-parser.js.

They are also the fastest and least flakey of our self-tests, hmm :)

The more code we can write to be unit-testable the better!